### PR TITLE
.github: update issue template

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

### What problem does this PR solve?

Problem Summary:

Challenge Program is ended and the entry causes only confusion.

Development Task is actually about enhancement and we setup an enhancement type, keep the word in consistency.

### Release note

- No release note.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/development-task.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
-name: "\U0001F680 Development Task"
-about: As a TiDB developer, I want to record a development task.
+name: "\U0001F680 Enhancement"
```
Comment (disksing) [inline .github/ISSUE_TEMPLATE/development-task.md:2]: do we need to update file name?

Comment (tisonkun) [reply .github/ISSUE_TEMPLATE/development-task.md:2]: Make sense.

**Review by disksing — APPROVED** [review]

**Review by jackysp — APPROVED** [review]

**Pull request merged**